### PR TITLE
Move file stageing

### LINKED ISSUE #9: Code in buildlib that needs to be in buildnml
The following code and perhaps more needs to be in the buildnml and not the buildlib.   https://github.com/ESCOMP/WW3_interface/blob/main/cime_config/buildlib#L105
The reason is that for a hybrid or branch run or in a ERR test the buildlib will not be run and therefore file mod_def.ww3 will not be present in the run directory.   The buildlib script is only intended to build the executable, file staging should be done in the buildnml script.

### PR DESCRIPTION
**Pull request by jedwards4b**
Move file staging from buildlib to buildnml.  
Tested using ERR_Ld5.ne30pg3_t232_wg37.1850_CAM70%LT_CLM60%BGC-CROP_CICE_MOM6_MOSART_DGLC%NOEVOLVE_WW3.derecho_intel

Fixes #9 

Mvertens would like to review @alperaltuntas can you add her?

### CODE REVIEW

**Review by alperaltuntas — COMMENTED** [review]
The `ww3_grid` executable gets built at the `buildlib` stage. If file staging gets moved from `buildlib` to `buildnml` and if the user attempts to, say, run `./preview_namelists` before running `./case.build` this change will lead to a failure due to the absence of `ww3_grid` executable, I think.

**Comment by jedwards4b** [comment]
See buildnml line 222.  Running buildnml before buildlib will print a warning about the missing executable but will no longer fail.

**Comment by alperaltuntas** [comment]
I am confused about the "resolve git issue" commit that was just pushed. It appears to change the WW3 submodule hash (why?) and when I click on the new submodule, GitHub displays: "This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository."

**Comment by jedwards4b** [comment]
I tried to undo that but had the same problem.   I don't think that it matters because the submodule will need to be updated again when this is merged.

**Comment by alperaltuntas** [comment]
The issue persists. When I create a test from scratch, buildnml doesn't generate or copy the ww3.mod_def file, causing the run to fail.

File: cime_config/buildnml [hunk 1]
```diff
@@ -179,6 +179,64 @@ def buildnml(case, caseroot, compname):
...
+
+        # Copy ww3_inp and other needed info to ww3_moddef_create directory
+        input_dir = case.get_value("WW3_GRID_INP_DIR")
+        files = os.listdir(input_dir)
+        for filename in files:
+            if not os.path.isfile(os.path.join(output_dir, filename)):
+                safe_copy(os.path.join(input_dir, filename), os.path.join(output_dir, filename))
+                
+        # Create mod_def file using ww3_grid and the grid_input files
+        run_cmd("ww3_grid", from_dir=output_dir)
```
Comment (alperaltuntas) [inline cime_config/buildnml:217]: Modifying this line as follows should fix this PR:

```run_cmd("./ww3_grid > mod_def.ww3.log", from_dir=output_dir)```

**Review by alperaltuntas — CHANGES_REQUESTED** [review]

File: cime_config/buildnml (hunk 1 above)
Comment (alperaltuntas) [reply cime_config/buildnml:217]: Better yet, I suggest replacing lines 218-223 with the following:

```
        if os.path.isfile(os.path.join(output_dir,"ww3_grid")): 
            run_cmd("./ww3_grid > mod_def.ww3.log", from_dir=output_dir)
            if not os.path.isfile(os.path.join(output_dir,"mod_def.ww3")):
                raise RuntimeError("mod_def.ww3 was not created, check mod_def.ww3.log for errors.")
            shutil.move(os.path.join(output_dir,"mod_def.ww3"), os.path.join(rundir, "mod_def.ww3"))
        else:
            logger.warning("ww3_grid file not found. The mod_def.ww3 file will be created after the build phase.")
```

Comment (alperaltuntas) [reply cime_config/buildnml:217]: Actually, even these changes don't fix the issue with TEST runs. It appears like, when you run a TEST case, the buildnml script doesn't get called in between build and run phases. And so the mod_def file never gets created in a TEST run.

Comment (jedwards4b) [reply cime_config/buildnml:217]: I have found the toggle to skip the pnl in tests and am testing a change to unset this when ww3 is in the compset.

**Comment by jedwards4b** [comment]
I've added a fix to my cime branch in https://github.com/ESMCI/cime/pull/4689
That branch will require other components to be updated as well, so here is the specific change that you can try in your current sandbox:

```
diff --git a/CIME/SystemTests/system_tests_common.py b/CIME/SystemTests/system_tests_common.py
index 70091aa29..a86483ffd 100644
--- a/CIME/SystemTests/system_tests_common.py
+++ b/CIME/SystemTests/system_tests_common.py
@@ -399,7 +399,13 @@ class SystemTestsCommon(object):
         """
         success = True
         start_time = time.time()
-        self._skip_pnl = skip_pnl
+        wav_comp = self._case.get_value("COMP_WAV")
+        # WW3 requires pnl to be run again after the build phase.
+        if wav_comp and wav_comp == "ww3":
+            self._skip_pnl = False
+        else:
+            self._skip_pnl = skip_pnl
+        
         try:
             self._resetup_case(RUN_PHASE)
             do_baseline_ops = True

```

**Comment by alperaltuntas** [comment]
Just confirming that this CIME mod, along with the changes I suggest above, fixed the issue.

**Comment by jedwards4b** [comment]
I've added you to my fork, please feel free to push your additional changes.

**Comment by alperaltuntas** [comment]
Done.

**Review by alperaltuntas — APPROVED** [review]

**Comment by alperaltuntas** [comment]
@mvertens, do you have any further comments for this PR? Otherwise, I'll merge this PR and create a tag.

**Review by mvertens — APPROVED** [review]
Sorry for the delay in my approving this. Looks good!

**Comment by mvertens** [comment]
Closing since wrong branch to merge to.

**Comment by mvertens** [comment]
Sorry about that. I was in the wrong PR>

**Comment by klindsay28** [comment]
I got bit by #9 using `create_clone --keepexe` in `cesm3_0_beta04`.
Is there anything holding up merging this fix?

**Comment by jedwards4b** [comment]
Maybe just confusion about who should do the merge?  I was expecting Alper to do it.

**Pull request merged**